ParU/Source/paru_internal.hpp

```diff
@@ -318,7 +316,7 @@ inline int32_t control_nthreads (ParU_Control *Control)
     }
     else
     {
-        return (MIN (max_threads, Control->paru_max_threads)) ;
+        return std::min(max_threads, static_cast<int32_t>(Control->paru_max_threads));
```

Comment:
That probably doesn't make a big difference here. But to avoid wrapping by the cast to a smaller type, it is generally safer to cast to the larger type in comparisons:
``` suggestion
        return std::min(static_cast<int64_t>(max_threads), Control->paru_max_threads);
```

Reply:
You are absolutely right. The wrapping occurs in any case. So, this change doesn't make any difference.
The only (very weak) advantage might be that if the return type is changed to `int64_t` at some point in the future, this line wouldn't need to be changed.
On the other hand, 32-bit comparisons might be slightly faster than 64-bit comparisons on 32-bit platforms. So, maybe best leave it as it is. 👍